### PR TITLE
adapt redis4net to our logstash setup

### PR DESCRIPTION
**Pull request by Oggmonster**
This PR 
- Adapt the redis4net (log4net appender) to our logstash setup - it first queues up log messages in redis before it gets picked up by logstash.

- nuget package available via gemfury

### CODE REVIEW

File: src/redis4net/LogMessage.cs [hunk 1]
```diff
@@ -114,25 +132,22 @@ public string Line
...
-			set
-			{
-				if (!this.ContainsKey("timestamp"))
-					this.Add("timestamp", value.ToUnixTimestamp().ToString(CultureInfo.InvariantCulture));
-				else
-					this["timestamp"] = value.ToUnixTimestamp().ToString(CultureInfo.InvariantCulture);
-			}
-		}
-	}
+                return (string)this["time"]; ;
```
Comment (markusahlstrand) [inline src/redis4net/LogMessage.cs]: A bonus ; ?

File: src/redis4net/Redis/IConnectionFactory.cs [hunk 2]
```diff
@@ -9,17 +9,21 @@ public class ConnectionFactory : IConnectionFactory
 		private static readonly object Lock = new object();
 
 		private readonly string _hostname;
-		private readonly int _portNumber;
+        private readonly string _password;
+        private readonly int _portNumber;
+        private readonly bool _enableSSL;
```
Comment (markusahlstrand) [inline src/redis4net/Redis/IConnectionFactory.cs]: Is the indenting a bit off here or is it just the formatting in github?

**Review by markusahlstrand — COMMENTED** [review]
Looks good! Two questions about the formatting

File: src/redis4net/LogMessage.cs (hunk 1 above)
Comment (Oggmonster) [reply src/redis4net/LogMessage.cs]: whoops!

File: src/redis4net/Redis/IConnectionFactory.cs (hunk 2 above)
Comment (Oggmonster) [reply src/redis4net/Redis/IConnectionFactory.cs]: yes, it looks a bit weird here - looks ok in visual studio though - but i can do an extra check

Comment (marcusstenbeck) [reply src/redis4net/Redis/IConnectionFactory.cs]: Looks like the first one is tabs and the second one is spaces.

File: src/redis4net/Layout/LogMessageLayout.cs [hunk 3]
```diff
@@ -207,7 +210,22 @@ private string GetValueFromPattern(LoggingEvent loggingEvent, string pattern)
...
+                {"ALL", 10 },
+                {"DEBUG", 20 },
+                {"INFO", 30 },
+                {"WARN", 40 },
+                {"ERROR", 50 },
+                {"FATAL", 60 },
+                {"OFF", 0 }
+            };
+            return levels.ContainsKey(level.Name) ? levels[level.Name] : 10;
+        }
```
Comment (ioanaRo) [inline src/redis4net/Layout/LogMessageLayout.cs]: magic numbers ?:D

File: src/redis4net/Redis/IConnectionFactory.cs (hunk 2 above)
Comment (Oggmonster) [reply src/redis4net/Redis/IConnectionFactory.cs]: yes - i see it in another editor. will fix

**Review by ioanaRo — COMMENTED** [review]
just a few magic numbers and strings here and there

File: src/redis4net/Layout/LogMessageLayout.cs [hunk 4]
```diff
@@ -207,7 +210,22 @@ private string GetValueFromPattern(LoggingEvent loggingEvent, string pattern)
 			}
 		}
 
-		private static long GetSyslogSeverity(Level level)
+        private static int GetLevelAsNumber(Level level)
+        {
+            var levels = new Dictionary<string, int>
```
Comment (marcusstenbeck) [inline src/redis4net/Layout/LogMessageLayout.cs]: Does it matter that this is created on each run? Couldn't it just as well be a constant? Maybe it's an OK thing in the C# world?

Comment (Oggmonster) [reply src/redis4net/Layout/LogMessageLayout.cs]: good catch. i'll move it outside.

**Pull request merged**